### ISSUE TITLE
Trouble getting schedule to work on mobile?

### ISSUE THREAD

**Issue by samaguilar** [bibekg/react-schedule-selector#25]
I'm having trouble getting the component to work on mobile, namely the ability to touch and drag over the tiles. I believe it has something to do with the touchstart Event Listener. I'm wondering if anyone else has struggled with this before and if there is a quick fix for it.

**Comment by bibekg** [comment]
Hm, which device/browser are you using? It seems to be working for me on my iPhone via Safari.

**Comment by samaguilar** [comment]
The demo happens to work flawlessly on mobile. For clarification, this issue seems to only occur on newly created projects using the package. I'll add the ScheduleSelector component to my new project, copy the same props/code from the demo, and still face the same problem (unable to click and drag over cells on mobile).

**Comment by bibekg** [comment]
Hm, could you share a code example where you run into this issue? Perhaps on something like codepen?

**Comment by samaguilar** [comment]
Sure thing! I've made a CodeSandbox with the package installed.
https://codesandbox.io/s/hardcore-nobel-jfu7i?file=/src/App.js

**Comment by bibekg** [comment]
Thanks for putting that together. I was able to reproduce the issue on my iPhone viewing your CodeSandbox example. However, even after making my own local sample app markup identical to your example, ie.
```jsx
<div className="App">
  <h1>Hello CodeSandbox</h1>
  <ScheduleSelector
    selection={this.state.schedule}
    numDays={5}
    minTime={8}
    maxTime={22}
    hourlyChunks={2}
    onChange={this.handleDateChange}
  />
</div>
```
I was still able to successfully select on mobile. This leads me to wonder if it's an issue with the way touch events get propagated through the `iframe` that CodeSandbox puts the app in.

**Comment by bibekg** [comment]
After doing a bit more in-depth debugging, I realized the problem stems from [Styled Component's deprecation of `innerRef`](https://styled-components.com/docs/api#deprecated-innerref-prop). A short-term solution is, if possible, to use a version of Styled System prior to v4. There's an active PR (#16) that uses the `ref` property and enables the use of the newest Styled Components. I'll work on getting that merged in and deployed in a new version.

**Comment by bibekg** [comment]
Just verified the fix from #29 makes your CodeSandbox example work. https://codesandbox.io/s/exciting-lamport-e34ec?file=/src/App.js

Let me know if it's all good on your end and I'll close this issue out.

**Comment by samaguilar** [comment]
Yes! Mobile is working on my end now. Really appreciate it!
